versioned/persist/serialize/src/main/java/org/projectnessie/versioned/persist/serialize/ProtoSerialization.java

```diff
@@ -287,6 +293,126 @@ public static Key protoToKey(AdapterTypes.Key key) {
+      case NONE:
+        c = AttachmentValue.Compression.NONE;
+        break;
+      case DEFLATE:
+        c = AttachmentValue.Compression.DEFLATE;
+        break;
+      case GZIP:
+        c = AttachmentValue.Compression.Gzip;
+        break;
+      default:
+        throw new IllegalArgumentException(
+            "Unsupported compression " + contentAttachment.getCompression());
+    }
+    AttachmentValue.Builder builder =
+        AttachmentValue.newBuilder().setCompression(c).setData(contentAttachment.getData());
+    if (contentAttachment.getVersion() != null) {
+      builder.setVersion(contentAttachment.getVersion());
+    }
+    return builder.build();
+  }
+
+  public static ContentAttachmentKey attachmentKey(AttachmentKey key) {
+    return ContentAttachmentKey.builder()
+        .contentId(ContentId.of(key.getContentId().getId()))
+        .objectType(key.getObjectType())
+        .objectId(key.getObjectId())
+        .build();
+  }
+
+  public static ContentAttachment attachmentContent(AttachmentKey key, AttachmentValue value) {
+    return attachmentContent(
+        ContentAttachment.builder()
+            .contentId(ContentId.of(key.getContentId().getId()))
+            .objectType(key.getObjectType())
+            .objectId(key.getObjectId()),
+        value);
+  }
+
+  public static ContentAttachment attachmentContent(
+      ContentAttachmentKey key, AttachmentValue value) {
+    return attachmentContent(
+        ContentAttachment.builder()
+            .contentId(ContentId.of(key.getContentId().getId()))
+            .objectType(key.getObjectType())
+            .objectId(key.getObjectId()),
+        value);
+  }
+
+  private static ContentAttachment attachmentContent(
+      ImmutableContentAttachment.Builder builder, AttachmentValue value) {
+    Compression c = compressionFromAttachmentValue(value);
+    builder.compression(c).data(value.getData());
+    if (value.hasVersion()) {
+      builder.version(value.getVersion());
+    }
+    return builder.build();
+  }
+
+  private static Compression compressionFromAttachmentValue(AttachmentValue value) {
+    Compression c;
+    switch (value.getCompression()) {
+      case NONE:
+        c = Compression.NONE;
+        break;
+      case DEFLATE:
+        c = Compression.DEFLATE;
+        break;
+      case Gzip:
+        c = Compression.GZIP;
+        break;
+      default:
+        throw new IllegalArgumentException("Unsupported compression " + value.getCompression());
+    }
+    return c;
+  }
+
+  public static String attachmentKeyAsString(ContentAttachmentKey key) {
+    return String.format(
+        "%s::%s::%s", key.getContentId().getId(), key.getObjectType(), key.getObjectId());
+  }
+
+  public static String attachmentKeyAsString(AttachmentKey key) {
+    return String.format(
+        "%s::%s::%s", key.getContentId().getId(), key.getObjectType(), key.getObjectId());
+  }
+
+  public static AttachmentKey attachmentKeyFromString(String key) {
```

Comment:
Should the methods that provide a construction of the `AttachmentKey` be in the `AttachmentKey.java` file?

Reply:
Tricky - `AttachmentKey` is a generated protobuf class